internal/wasm/func_validation.go

```diff
@@ -1464,6 +1464,9 @@ func (m *Module) validateFunctionWithMaxStackValues(
 				valueTypeStack.push(p)
 			}
 		} else if op == OpcodeEnd {
+			if len(controlBlockStack) == 0 {
```

Comment:
found a bug actually :D

Reply:
oh maybe I need to add regression test case

Reply:
yeah thanks I should have nagged you, but you self-nagged. excellent